aws/checksum.go

```diff
@@ -0,0 +1,39 @@
+package aws
+
+// RequestChecksumCalculation controls request checksum calculation workflow
+type RequestChecksumCalculation string
+
+const (
+	// RequestChecksumCalculationWhenSupported indicates request checksum should be calculated if modeled
+	RequestChecksumCalculationWhenSupported RequestChecksumCalculation = "whenSupported"
```

Comment:
Idiomatically these should be `type <x> int`, I know we've deviated from that somewhat in the past, but I'd like us to course-correct.